crates/csharp/src/function.rs

```diff
@@ -1033,19 +1039,72 @@ impl Bindgen for FunctionBindgen<'_, '_> {
             Instruction::Malloc { .. } => unimplemented!(),
 
             Instruction::GuestDeallocate { .. } => {
-                uwriteln!(self.src, r#"Console.WriteLine("TODO: deallocate buffer for indirect parameters");"#);
+                // the original alloc here comes from cabi_realloc implementation (wasi-libc in .net)
+                uwriteln!(self.src, r#"NativeMemory.Free((void*){});"#, operands[0]);
```

Comment:
Which test would be executing this new line ?

Reply:
The runtime tests that exercises this is https://github.com/bytecodealliance/wit-bindgen/tree/main/tests/runtime/resource_aggregates